hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/InformixDialect.java

```diff
@@ -269,19 +276,41 @@ public void initializeFunctionRegistry(FunctionContributions functionContributio
 		functionFactory.initcap();
 		functionFactory.yearMonthDay();
 		functionFactory.ceiling_ceil();
-		functionFactory.concat_pipeOperator();
+		functionFactory.concat_pipeOperator( SqlAstNodeRenderingMode.INLINE_ALL_PARAMETERS );
 		functionFactory.ascii();
 		functionFactory.char_chr();
 		functionFactory.addMonths();
 		functionFactory.monthsBetween();
 		functionFactory.stddev();
 		functionFactory.variance();
-		functionFactory.locate_positionSubstring();
-
-		//coalesce() and nullif() both supported since Informix 12
+		functionFactory.bitLength_pattern( "length(?1)*8" );
 
+		if ( getVersion().isBefore( 12 ) ) {
+			functionContributions.getFunctionRegistry().register(
+					"coalesce",
+					new NvlCoalesceEmulation( SqlAstNodeRenderingMode.INLINE_ALL_PARAMETERS )
```

Comment:
Are you sure `NO_PLAIN_PARAMETER` won't work? Inlining parameters really is the last resort and shouldn't be done lightly.